test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTrimTest.java

```diff
@@ -174,6 +182,68 @@ public void testSuccessiveCheckpointTrim() throws Exception {
                 .isEqualTo("a" + (nCheckpoints - 1));
     }
 
+    /**
+     * Verify that the streaming interface can be consumed directly and that
+     * {@link TrimmedException} is being thrown when linearizable history is lost.
+     */
+    @Test
+    public void rawStreamConsumer() {
+        final int BATCH_SIZE = 10;
+        final int CHECKPOINT_SIZE = 3;
+        final String CHECKPOINT_AUTHOR = "Author";
+        final String tableName = "test";
+        final CorfuTable<Integer, Integer> map = getDefaultRuntime().getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<Integer, Integer>>() {})
+                .setStreamName(tableName)
+                .open();
+
+        final MultiCheckpointWriter<CorfuTable> mcw = new MultiCheckpointWriter();
+        mcw.addMap(map);
+
+        IntStream.range(0, BATCH_SIZE).forEach(idx -> map.put(idx, idx));
+
+        // Insert a checkpoint
+        Token checkpointAddress = mcw.appendCheckpoints(getRuntime(), CHECKPOINT_AUTHOR);
+
+        IntStream.range(0, BATCH_SIZE).forEach(idx -> map.put(idx, idx));
+
+        // Trim the log in between the checkpoints
+        trim(checkpointAddress);
+
+        CorfuRuntime newRuntime = getNewRuntime(getDefaultNode()).connect();
+        Map<Integer, Integer> newMap = newRuntime.getObjectsView().build()
+                .setTypeToken(new TypeToken<SMRMap<Integer, Integer>>() {
+                })
+                .setStreamName(tableName)
+                .open();
+
+        IStreamView s = newRuntime.getStreamsView().get(CorfuRuntime.getStreamID(tableName));
+        s.seek(BATCH_SIZE + CHECKPOINT_SIZE);
+        // Seek beyond the last trimmed address.
+        // The first call to remainingUpTo() will load the checkpoint, and the
+        // second one will fetch the actual data.
+        Assertions.assertThat(Stream.of(s.remainingUpTo(Long.MAX_VALUE), s.remainingUpTo(Long.MAX_VALUE))
+                .map(List::size).mapToInt(Integer::intValue).sum())
+                .isEqualTo(BATCH_SIZE);
+
+        trim(mcw.appendCheckpoints(getRuntime(), CHECKPOINT_AUTHOR));
+        IntStream.range(0, BATCH_SIZE).forEach(idx -> newMap.put(idx, idx));
+        Assertions.assertThatThrownBy(() -> s.remainingUpTo(Long.MAX_VALUE))
+                .isInstanceOf(TrimmedException.class);
+    }
+
+    /**
+     * Given the token, trim the address-space at {@link Token#getSequence()}.
+     *
+     * @param token point at which to trim the address space.
+     */
+    private void trim(Token token) {
+        getRuntime().getAddressSpaceView().prefixTrim(token);
+        getRuntime().getAddressSpaceView().gc();
+        getRuntime().getAddressSpaceView().invalidateServerCaches();
+        getRuntime().getAddressSpaceView().invalidateClientCache();
+    }
+
```

Comment:
I see `testCheckpointTrim` and `testSuccessiveCheckpointTrim` use these four lines to trim.
Can we replace them with trim()?

Reply:
Sounds good.